stdlib/Test/src/Test.jl

```diff
@@ -1389,6 +1389,28 @@ function _inferred(ex, mod, allow = :(Union{}))
     end)
 end
 
+"""
+    isequal_nolines(a, b)
+
+Compare two Expr trees for equality, ignoring LineNumberNodes, so that otherwise identical
+Exprs defined in two different locations will still compare equal.
+
+# Example
+```jldoctest
+julia> a = quote 1 end;  # Line: #= REPL[1]:1 =#
+
+julia> b = quote 1 end;  # Line: #= REPL[2]:1 =#
+
+julia> Test.isequal_nolines(a, b)
+true
+```
+
+See also: [`isapprox`](@ref) for _numerical_ approximate equality
+"""
+function isequal_nolines(a, b)
+    Base.remove_linenums!(a) == Base.remove_linenums!(b)
```

Comment:
This should `copy` its arguments so the predicate overall is non-mutating.

Reply:
Oops, duh, thanks.

`deepcopy()` or `copy()`? I feel like `deepcopy`, but i still don't _really_ understand the difference b/w them.

Reply:
`copy` on a container makes a new container with the same elements. So copy of an array allocates exactly one new array. Exprs are more of a tree, so it copies the tree structure (Exprs and all sub-Exprs) but keeps the "leaves". So I think `copy` is correct here.

`deepcopy` attempts to copy everything reachable from an object, as if you serialized the whole thing to disk and read it back in. `deepcopy` is best avoided, since it violates abstraction.

Reply:
```suggestion
    a = Base.remove_linenums!(copy(a))
    b = Base.remove_linenums!(copy(b))
    return a == b
```